lib/formtools/overview.js

```diff
@@ -0,0 +1,183 @@
+
+var async = require('async'),
+    cellRenderers = require('./renderers-cell');
+
+/**
+ * Set unique tabId to each tab in overview.body
+ * @param {Array} sections overview body array
+ */
+function setTabId (sections) {
+
+    sections.forEach(function (section, sectionIndex) {
+
+        if (!Array.isArray(section)) {
+            return;
+        }
+
+        section.forEach(function (tab, tabIndex) {
+
+            // Replace white space from tab.lable with "-", make it lowercase and add the index number at the end.
+            // sectionIndex and tabIndex number are added at the end to make the tabID unique in case when more then one tab have the same label (title)
+            tab.tabId = (tab.label).replace(/\s+/g, '-').toLowerCase() + '-' + sectionIndex + '-' + tabIndex;
+
+        });
+
+    });
+
+}
+
+/*
+ * Update section.fields to an array of objects with label and value keys. Label should contain the field label and value should contain the field value.
+ * e.g ({ label: '', value: '' })
+ * @param  {Object}   record   Overview record object
+ * @param  {Object}   model    Model object
+ * @param  {Object}   section  Section Object with its label and an array of fields (e.g { label: '', fields: [] })
+ * @param  {Function} callback Callback function
+ * @return {[object]}          Return object: { label: '', fields: [ { label: '', value: '' }, ... ] } to callback
+ */
+function renderFields(record, model, section, callback) {
+
+    // filter out invalid fields from section
+    var fields = section.fields.filter(function (field) {
+        return model.linz.formtools.form[field];
+    });
+
+    async.mapSeries(fields, function (field, cb) {
+
+        var formField = model.linz.formtools.form[field],
+            renderer = cellRenderers[formField.type] ? formField.type : 'text';
+
+        cellRenderers[renderer](record[field], record, field, model, function (err, value) {
+
+            if (err) {
+                return cb(err);
+            }
+
+            return cb(null, {
+                label: model.linz.formtools.labels[field],
+                value: value
+            });
+
+        });
+
+    }, function (err, result) {
+        return callback(err, { label: section.label, fields: result } );
+    });
+
+}
+
+/**
+ * Recursively loops through provided section to:
+ * (1) Render body if section.body is a function (e.g. { label: '', body: '' })
+ * (2) Populate field labels, if section.fields is defined (by converting section.fields into an array of objects. e.g. { label: '', fields: [ { label: '', value: '' }, ... ] })
+ * (3) If section.body is an array, recursively repeats step (1) and (2) (e.g. { label: '', body: [] })
+ * @param  {[Object]}   req    Request object
+ * @param  {[Object]}   res    Response onject
+ * @param  {[Object]}   record Overview record
+ * @param  {[Object]}   model  Model object
+ * @param  {Object or Array}   section from overvie DSL which can be an Array or an Object
```

Comment:
Update comment to below:
`section from overview body array which can be an Array or an Object`